lightning/src/ln/channelmanager.rs

```diff
@@ -10389,14 +10419,22 @@ where
 				}
 			},
 			#[cfg(async_payments)]
-			OffersMessage::StaticInvoice(_invoice) => {
-				match responder {
-					Some(responder) => {
-						responder.respond(OffersMessage::InvoiceError(
-								InvoiceError::from_string("Static invoices not yet supported".to_string())
-						))
-					},
+			OffersMessage::StaticInvoice(invoice) => {
+				let responder = match responder {
+					Some(responder) => responder,
 					None => return ResponseInstruction::NoResponse,
+				};
+				let payment_id = match _payment_id {
+					Some(id) => id,
+					None => {
+						return responder.respond(OffersMessage::InvoiceError(
+							InvoiceError::from_string("Unrecognized invoice".to_string())
+						))
+					}
+				};
```

Comment:
Let's not respond if we don't recognize the payment id. I have an update in #3139 to do the same elsewhere.